internal/testhelper/assert.go

```diff
@@ -0,0 +1,36 @@
+package testhelper
```

Comment:
@bzwei I think we should use this repo https://github.com/stretchr/testify it has a nice mock and assertion interface that we can directly use. There are lot of examples on how to use it. There is also a mockery repo which can be used in conjunction with testify to generate mock stubs for a given interface, that generate the mocks into a separate directory and we can just use those generated mocks.